server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1541,8 +1624,10 @@ private void addDeleteListener(String deleteUUID, ActionListener<Void> listener)
     public Version minCompatibleVersion(Version minNodeVersion, RepositoryData repositoryData, @Nullable Collection<SnapshotId> excluded) {
         Version minCompatVersion = minNodeVersion;
         final Collection<SnapshotId> snapshotIds = repositoryData.getSnapshotIds();
-        for (SnapshotId snapshotId : snapshotIds.stream().filter(excluded == null ? sn -> true : sn -> excluded.contains(sn) == false)
-                .collect(Collectors.toList())) {
+        for (SnapshotId snapshotId : snapshotIds) {
+            if (excluded != null && excluded.contains(snapshotId)) {
+                continue;
+            }
```

Comment:
Sneaked in a non-style change because `for (.... .collect(Collectors.toList())` is a waste